### DISCUSSION TITLE
Creating custom elements to display native rendering contexts

### DISCUSSION THREAD

**Discussion by BlueNebulaDev** [servo/servo#28540] [Q&A] [answered]
I would like to define a new tag `<mpv>` to [embed the mpv media player](https://github.com/mpv-player/mpv-examples/blob/master/libmpv/README.md#methods-of-embedding-the-video-window) in an HTML page rendered by servo.

I'm guessing that I would need to create a new tag, similar to `<video>` or `<canvas>`, that calls mpv's functions to render the element and handles all the properties and events accordingly.

Is it possible to extend servo this way? Are there any examples showing how to do this?

**Comment by jdm** [comment, marked as answer]
It is not currently possible to extend Servo in this way without forking the engine and duplicating the video-related functionality from https://github.com/servo/servo/blob/master/components/script/dom/htmlmediaelement.rs .

**Reply by jdm** [reply]
Servo _is_ set up to support integrating different media backends than gstreamer through https://github.com/servo/media, which might be a more realistic route.
